plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -137,6 +137,10 @@ export class ActionMatcher {
         this.statsd = statsd
     }
 
+    public hasWebhooks(teamId: number): boolean {
+        return Object.keys(this.actionManager.getTeamActions(teamId)).length > 0
+    }
```

Comment:
Perhaps this should be moved one level deeper, into `actionManager` and the code here can just direct to `this. actionManager.teamHasWebhooks(teamId)`?

Also, shouldn't this check the `slack_webhook` field?

Reply:
actionManager is private and this was available in the call location, so that's why it's here. 
If you look at the code for what the action matcher does you'll see that we return based on a cache that gets re-evaluated every 5min and at the start. What it does is get all the actions where post_to_slack is true on the action or there's a hook for this action. We could additionally filter for where the team.slack_webhook is set for `post_to_slack` but that would mean more code changes (need to split out the hooks part somehow), so I opted not to do it, assuming folks don't randomly check the post_to_slack checkbox for no good reason.

Reply:
I think the naming is a bit confusing, because originally (pre-#10414) we _did_ just fetch all the actions. Then that was made smarter, but the method/function names don't reflect this. E.g. `getTeamActions` could be `getHookEnabledTeamActions` (verbose but clearer)